src/main/java/spireCafe/util/CafeUtil.java

```diff
@@ -0,0 +1,143 @@
+                    m.setAccessible(true);
+                    canSpawnMethod = m;
+                    break;
+                }
+            }
+            canSpawnMethods.put(clz, canSpawnMethod);
+        }
+        try {
+            return canSpawnMethod == null || (boolean)canSpawnMethod.invoke(null);
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    public static HashMap<String, Class<? extends AbstractCafeInteractable>> getAllValidInteractableByType(Class<? extends AbstractCafeInteractable> clz) {
+        return Anniv7Mod.interactableClasses.entrySet().stream()
+        .filter(e -> clz.isAssignableFrom(e.getValue()))
+        .filter(e -> !Anniv7Mod.currentRunSeenInteractables.contains(e.getKey()))
+        .filter(e -> canSpawn(e.getValue()))
+        .collect(Collectors.toMap(
+            e -> e.getKey(),
+            e -> e.getValue(),
+            (oldValue, newValue) -> oldValue,
+            HashMap::new));
+    }
+
+    public static HashMap<String, Class<? extends AbstractCafeInteractable>> getValidAttractions() {
+         return getAllValidInteractableByType(AbstractAttraction.class);
+    }
+    public static HashMap<String, Class<? extends AbstractCafeInteractable>> getValidBartenders() {
+         return getAllValidInteractableByType(AbstractBartender.class);
+    }
+    public static HashMap<String, Class<? extends AbstractCafeInteractable>> getValidMerchants() {
+         return getAllValidInteractableByType(AbstractMerchant.class);
+    }
+    public static HashMap<String, Class<? extends AbstractCafeInteractable>> getValidPatrons() {
+         return getAllValidInteractableByType(AbstractPatron.class);
+    }
+
+    public static List<String> getValidAttractionIDs() {
+        return getValidAttractions().keySet().stream().collect(Collectors.toList());
+    }
+    public static List<String> getValidBartenderIDs() {
+        return getValidBartenders().keySet().stream().collect(Collectors.toList());
+    }
+    public static List<String> getValidMerchantIDs() {
+        return getValidMerchants().keySet().stream().collect(Collectors.toList());
+    }
+    public static List<String> getValidPatronIDs() {
+        return getValidPatrons().keySet().stream().collect(Collectors.toList());
+    }
+
+    public static void markInteractableAsSeen(String ID){
+        Anniv7Mod.currentRunSeenInteractables.add(ID);
+    }
+
+    public static void markInteractableAsSeen(AbstractCafeInteractable interactable) {
+        Anniv7Mod.currentRunSeenInteractables.add(interactable.id);
+    }
+
+    public static Class <? extends AbstractCafeInteractable> getInteractableClass(String ID) {
+        return Anniv7Mod.interactableClasses.get(ID);
+    }
+
+    public static AbstractCafeInteractable createInteractable(Class<? extends AbstractCafeInteractable> clz, float x, float y) {
```

Comment:
Like `canSpawn` above, you can make the method this is based on from `CafeRoom` public and change this to call that method.